client/source/components/RelatedProducts/RelatedList.jsx

```diff
@@ -24,16 +27,51 @@ const RelatedList = function CreateRelatedList() {
       .catch((error) => { console.error(error); });
   }, [prodID]);
 
+  // HANDLERS
+  const handleRightClick = () => {
+    flushSync(() => {
+      if (index < relatedProducts.length - 1) {
+        setIndex(index + 1);
+      } else {
+        setIndex(0);
+      }
+    });
+    selectedRef.current.scrollIntoView({
+      behavior: 'smooth',
+      block: 'nearest',
+      inline: 'center'
+    });
+  };
+
+  const handleLeftClick = () => {
+    flushSync(() => {
+      if (index > 0) {
+        setIndex(index - 1);
+      } else {
+        setIndex(relatedProducts.length - 1);
+      }
+    });
+    selectedRef.current.scrollIntoView({
+      behavior: 'smooth',
+      block: 'nearest',
+      inline: 'center'
+    });
+  };
+
   return (
     <StyledRelatedList data-module-name="related products">
-      <h3>Related Products:</h3>
+      <h3 data-testid="heading">Related Products</h3>
       <div className="container">
         <StyledMediaScroll>
-          {/* eslint-disable-next-line max-len */}
-          {relatedProducts.map((relatedProdId) => <RelatedListCard key={relatedProdId} relatedProdId={relatedProdId} />)}
+          {/* eslint-disable-next-line max-len, react/jsx-wrap-multilines */}
+          {relatedProducts.map((relatedProdId, i) => <RelatedListCard
+            key={relatedProdId}
+            relatedProdId={relatedProdId}
+            ref={index === i ? selectedRef : null}
+          />)}
         </StyledMediaScroll>
-        <BiChevronLeftCircle className="left" />
-        <BiChevronRightCircle className="right" />
+        {index > 0 && <BiChevronLeftCircle data-testid="reactIcon" className="left" onClick={handleLeftClick} />}
```

Comment:
Currently you move by index, but since you show more than 1 item at a time you can click several times and not change anything visually. Maybe try scrolling a set offset based on the width of a rendered card?

Reply:
|| [ * ]  [  ]  [  ] || [  ]   index is 0     [ cards]    || screen width ||  

click the right button...

|| [  ]  [ * ]  [  ] || [  ]      nothing happens, since index 1 is already in view